tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/access/FieldAccessTest.java

```diff
@@ -127,6 +127,7 @@ public void testNegativeCases() {
         validateError(negativeResult, i++, "field access can only be used to access required fields or optional " +
                 "fields of non-nilable types, field 'z' is undeclared in record(s) 'CD' and type includes nil in " +
                 "record(s) 'BC'", 331, 17);
+        validateError(negativeResult, i, "undefined field 'id' in | '(AB|BC)'", 337, 5);
```

Comment:
IMO, It will be better if we can change the error msg into either `undefined field 'id' in union '(AB|BC)'` or `undefined field 'id' in type '(AB|BC)'`. WDYT @KavinduZoysa?

Reply:
Changed